source/Calamari.Aws/Deployment/Conventions/DeployAwsCloudFormationConvention.cs

```diff
@@ -865,5 +892,21 @@ private bool DisplayWarning(string errorCode, string message)
 
             return false;
         }
+
+        /// <summary>
+        /// The AmazonServiceException can hold additional information that is useful to include in
+        /// the log.
+        /// </summary>
+        /// <param name="exception">The exception</param>
+        private void HandleAmazonServiceException(AmazonServiceException exception)
+        {
+            if (exception.InnerException is WebException)
+            {
+                ((WebException) exception.InnerException).Response.GetResponseStream()
```

Comment:
Btw you can avoid the additional cast here if you wanted to by using pattern matching i.e.

```
private void HandleAmazonServiceException(AmazonServiceException exception)
        {
            if (exception.InnerException is WebException innerException)
            {
                (innerException.Response.GetResponseStream()
                    .Map(stream => new StreamReader(stream).ReadToEnd())
                    .Tee(message => DisplayWarning("AWS-CLOUDFORMATION-ERROR-0014",
                        "An exception was thrown while contacting the AWS API.\n" + message));
            }
        }
```